magento/pwa-studio
# PWA-2005 Contact Us page

## Description
Adding front-end support for contact-us functionality
- Refactored footer to be able to inject component link
- Added conditional link to footer when Contact Us is enabled
- Added new static route and page component
- Added Contact Us Form component + talon
- Added Cypress test coverage

## Related Issue







Closes PWA-2005

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature

- Use link in footer to navigate to contact us page
- Try to submit empty form. Should show errors on fields
- Enter all information but an invalid email address (omit `@`). Should submit and display a relavent error message
- Submit valid form. Should display a 'toast' message. Email configured in backend should receive an email.

Run Cypress test
```
yarn test:headless --url https://pr-3538.pwa-venia.com/ --spec src/tests/integrationTests/contactPage/userCanSubmitForm.spec.js
```

#### Test scenario(s) for any existing impacted features/areas
- Ensure existing footer links function correctly

#### Test scenario(s) for any Magento Backend Supported Configurations
- Disable Contact Us.
- Link should disappear from footer
- Visit /contact-us. Should see the 404 messaging

#### Is Browser/Device testing needed?
Yes, device testing is needed as page layout may be impacted on mobile

#### Any ad-hoc/edge case scenarios that need to be considered?
- Form should submit correctly with or without telephone input

## Checklist
